nightfall-client/src/services/commitment-storage.mjs

```diff
@@ -688,3 +688,67 @@ export async function findUsableCommitmentsMutex(
     findUsableCommitments(compressedZkpPublicKey, ercAddress, tokenId, _value, onlyOne),
   );
 }
+
+/**
+ * @function getCommitmentsByCompressedPkdList do the role of a service taking care of the
+ * business logic and of a repository doing the communication with the database for this
+ * use case.
+ * @param {string[]} compressedZkpPublicKey a list of compressedZkpPublicKey derivated from the user
+ * mnemonic coming from the SDK or Wallet.
+ * @returns all the commitments existent for this list of compressedZkpPublicKey.
+ * @author luizoamorim
+ */
+export async function getAllCommitmentsByCompressedZkpPublicKeyList(listOfCompressedZkpPublicKey) {
+  const connection = await mongo.connection(MONGO_URL);
+  const db = connection.db(COMMITMENTS_DB);
+  const allCommitmentsByListOfCompressedZkpPublicKey = await db
+    .collection(COMMITMENTS_COLLECTION)
+    .find({
+      'preimage.compressedZkpPublicKey': { $in: listOfCompressedZkpPublicKey },
+    })
+    .toArray();
+  return allCommitmentsByListOfCompressedZkpPublicKey;
+}
+
+/**
+ * @function getAllCommitments do the role of a service taking care of the
+ * business logic and of a repository doing the communication with the database for this
+ * use case.
+ * @returns all the commitments existent in this database.
+ * @author luizoamorim
+ */
+export async function getAllCommitments() {
+  const connection = await mongo.connection(MONGO_URL);
+  const db = connection.db(COMMITMENTS_DB);
+  const allCommitments = await db.collection(COMMITMENTS_COLLECTION).find({}).toArray();
+  return allCommitments;
```

Comment:
```suggestion
  return db.collection(COMMITMENTS_COLLECTION).find().toArray();
```
Haven't tested this, but pretty sure you don't need the curly braces `{}`

Reply:
![image](https://user-images.githubusercontent.com/73957838/179582295-74373036-e13a-4a55-abd4-d505a8a08269.png)
You're correct.